receiver/k8sclusterreceiver/e2e_test.go

```diff
@@ -58,17 +61,16 @@ func TestE2E(t *testing.T) {
 	defer shutdownSink()
 
 	testID := uuid.NewString()[:8]
-	collectorObjs := k8stest.CreateCollectorObjects(t, k8sClient, testID, "")
+	collectorObjs := k8stest.CreateCollectorObjects(t, k8sClient, testID, filepath.Join(".", "testdata", "e2e", "cluster-scoped", "collector"))
 
 	t.Cleanup(func() {
 		for _, obj := range append(collectorObjs) {
 			require.NoErrorf(t, k8stest.DeleteObject(k8sClient, obj), "failed to delete object %s", obj.GetName())
 		}
 	})
 
-	wantEntries := 10 // Minimal number of metrics to wait for.
+	wantEntries := expected.ResourceMetrics().Len() // Minimal number of metrics to wait for.
```

Comment:
Is this change needed here? In that case, I wonder if we should do any refactoring to the existing tests in a separate PR.